R/extract_data.R

```diff
@@ -0,0 +1,245 @@
+#' Extract data from traits.build database
+#' 
+#' @description Function to extract data from a traits.build database based on 
+#' any value(s) from any column in the traits, locations, contexts, methods, 
+#' taxa, taxonomic_updates, and contributors tables.
+#' The output a traits.build formatted database with all tables subset 
+#' based on the specified table, column (variable) and column value.
+#'
+#' @param database traits.build database
+#' @param table Table within a traits.build database
+#' @param col Column name within the specified table.
+#' @param col_value Value (of column, from with a table) that is used to subset database. This can be a single value or a vector. It includes partial string matches. 
+#'
+#' @return subset traits.build database
+#' @export
+#'
+#' @examples 
+#' \dontrun{
+#' extract_data(database = traits.build_database, table = "traits", 
+#' col = "trait_name", col_value = "leaf_area")
+#' }
+extract_data <- function(database, table, col, col_value) {
+  
+  database$contexts <- database$contexts %>% tidyr::separate_longer_delim(link_vals, delim = ", ")
+
+  database$contexts_tmp <- split(database$contexts, database$contexts$link_id)
```

Comment:
I don't see contexts_tmp being called anywhere else after being created here. Is this part needed? I might be missing something!

Reply:
@fontikar It is use in lines 38-71 below, to finish creating the individual context tables

Reply:
I wrote some code to reduce duplication from lines 58 - 71

It uses the := walrus operator to dynamically rename link_vals. 

```
rename_contexts_link_vals <- function(table, name) {
  table %>%
    dplyr::rename(!!rlang::quo_name(name) := link_vals)
}

purrr::map(database$contexts_tmp, 
    ~rename_contexts_link_vals(.x, .x$link_id |> unique()))
```

Reply:
Ended up using a variant of this, because this was giving a weird error about eval syntax